main/cache/cache.cc

```diff
@@ -13,7 +13,16 @@ unique_ptr<OwnedKeyValueStore> maybeCreateKeyValueStore(shared_ptr<::spdlog::log
     if (opts.cacheDir.empty()) {
         return nullptr;
     }
-    auto flavor = opts.lspExperimentalFastPathEnabled ? "experimentalfastpath" : "normalfastpath";
+
+    std::string flavor;
+    if (opts.print.isAutogen() && !opts.print.isAutogenPrintingDepDB()) {
+        flavor = "autogenemptymethodbodies";
+    } else if (opts.lspExperimentalFastPathEnabled) {
```

Comment:
Use a different cache flavor in empty method mode to avoid poisoning any existing caches.